### PR TITLE
register completed

### PR DESCRIPTION
**Pull request by mageexxcel**

### CODE REVIEW

File: src/app/client/components/Register.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    this.autorun.stop()
+  }
+  signup(){
+    let email = this.state.email.trim()
+    let name = this.state.name.trim()
+    let password = this.state.password.trim()
+    if(email == ""){
+      alert("Email required")
+    }else if(!/^\w+@[a-zA-Z_]+?\.[a-zA-Z]{2,3}$/.test(email)){
+      alert("Please enter a valid email")
```
Comment (arun-kumar-codes) [inline src/app/client/components/Register.js]: dont user alerts. show error message using material design compenent

File: src/app/client/components/Register.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+          conf_password : ''
+        })
+      }).catch( (error) => {
+        //console.log(error)
+          alert(error.toString())
+      })
+    }
+  }
+  goBack(){
+    alert("goBack triggered")
```
Comment (arun-kumar-codes) [inline src/app/client/components/Register.js]: this should go to login page

File: src/app/server/publications.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { Meteor } from 'meteor/meteor';
-import Tasks from 'TodoApp/collections/Tasks';
+import Tasks from 'app/collections';
 
 // This code only runs on the server
 Meteor.publish('tasks', function () {
```
Comment (arun-kumar-codes) [inline src/app/server/publications.js]: what is this?

File: src/app/collections/index.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { Mongo } from 'meteor/mongo'
+
+import '../methods/index'
+
+
+const Tasks = new Mongo.Collection("tasks");
```
Comment (arun-kumar-codes) [inline src/app/collections/index.js]: ??

File: src/app/client/reducers/register.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+export function register(state = Immutable.fromJS({}),action){
+	switch(action.type){
+		case ADD_USER:
+			state = Immutable.fromJS(action.payload)
+			console.log(state)
+			return _.clone(state)
+
+		default:
+			return _.clone(state)
```
Comment (arun-kumar-codes) [inline src/app/client/reducers/register.js]: u dont need to clone state in default

File: src/app/client/reducers/register.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+import { ADD_USER } from '../actions'
+
+
+
+export function register(state = Immutable.fromJS({}),action){
+	switch(action.type){
+		case ADD_USER:
+			state = Immutable.fromJS(action.payload)
+			console.log(state)
+			return _.clone(state)
```
Comment (arun-kumar-codes) [inline src/app/client/reducers/register.js]: this is not required. immutablejs already done clone internally its not needed again

File: src/app/client/reducers/register.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import Immutable from 'immutable';
+
+import { ADD_USER } from '../actions'
+
+
+
+export function register(state = Immutable.fromJS({}),action){
+	switch(action.type){
+		case ADD_USER:
+			state = Immutable.fromJS(action.payload)
```
Comment (arun-kumar-codes) [inline src/app/client/reducers/register.js]: change state as discussed

File: src/app/client/components/Register.js (hunk 1 above)
Comment (mageexxcel) [inline src/app/client/components/Register.js]: done

**Pull request merged**